utils/sampler/weighted_best.go

```diff
@@ -41,15 +41,11 @@ func (s *weightedBest) Initialize(weights []uint64) error {
 		totalWeight = newWeight
 	}
 
-	if totalWeight > math.MaxInt64 {
-		return errWeightsTooLarge
-	}
-
 	samples := []uint64(nil)
 	if totalWeight > 0 {
 		samples = make([]uint64, s.benchmarkIterations)
 		for i := range samples {
-			samples[i] = uint64(globalRNG.Int63n(int64(totalWeight)))
+			samples[i] = globalRNG.Uint64n(totalWeight - 1)
```

Comment:
Note the `-1` here now because `Uint64n` returns values in the range `[0, n]` whereas `Int63n` returned values in the range `[0, n)`